### ISSUE TITLE
Fix `pnpm run test run`

### ISSUE THREAD

**Issue by ST-DDT** [faker-js/faker#1635]
### Pre-Checks

- [X] Follow our [Code of Conduct](https://github.com/faker-js/faker/blob/next/CODE_OF_CONDUCT.md).
- [X] Read the [Contributing Guidelines](https://github.com/faker-js/faker/blob/next/CONTRIBUTING.md).
- [X] Read the [docs](https://fakerjs.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/faker-js/faker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a Faker issue and not related to a combination with another package.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/faker-js/faker/discussions) or join our [Discord Chat Server](https://chat.fakerjs.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.
- [ ] I am willing to provide a PR.

### Describe the bug

`pnpm run test run` fails due to tests timing out.

> test/locale-imports.spec.ts > locale imports > Internal tests to cover `src/locale/*.ts` > should be possible to directly import('../src/locale/af_ZA')
Error: Test timed out in 5000ms.
If this is a long-running test, pass a timeout value as the last argument or configure it globally with "testTimeout".

````txt
  ❯ test/locale-imports.spec.ts (232) 41438ms
   ❯ locale imports (232) 41437ms
     ✓ should be possible to directly require('@faker-js/faker/locale/af_ZA')
     ✓ should be possible to directly import('@faker-js/faker/locale/af_ZA') 1009ms
     ❯ Internal tests to cover `src/locale/*.ts` (2) 5035ms
       ✓ should be possible to directly require('../locale/af_ZA')
       × should be possible to directly import('../src/locale/af_ZA') 5027ms
     ✓ should be possible to directly require('@faker-js/faker/locale/ar')
     ✓ should be possible to directly import('@faker-js/faker/locale/ar') 4299ms
     ✓ Internal tests to cover `src/locale/*.ts` (2) 805ms
     ✓ should be possible to directly require('@faker-js/faker/locale/az')
     ✓ should be possible to directly import('@faker-js/faker/locale/az') 387ms
````

### Minimal reproduction code

- `pnpm run test run`

### Additional Context

These two **previously** worked:

- `pnpm run test`
- `pnpm run test run test/locale-imports.spec.ts`

### Environment Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (16) x64 Intel(R) Core(TM) i7-7820X CPU @ 3.60GHz
    Memory: 53.93 GB / 63.69 GB
  Binaries:
    Node: 18.12.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 8.16.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22621.819.0), Chromium (108.0.1462.42)
    Internet Explorer: 11.0.22621.1
```

### Which module system do you use?

- [ ] CJS
- [ ] ESM

### Used Package Manager

pnpm

**Comment by Shinigami92** [comment]
<img src="https://user-images.githubusercontent.com/7195563/210084474-014aa0d4-04b5-40f6-8ec0-82568740496c.png" width="300px"/>

I do not have timeouts anymore, but running the tests take tremendously huge amount of time (~20 min)
Looks like the `all_functional` and `locales` tests take >1k seconds
For me it feels like this is too long and we should try to bring them down to something faster

**Comment by ST-DDT** [comment]
This also seems to affect the normal tests: `pnpm run test`

Workaround:

`CI=true pnpm run test`
`CI=true pnpm run test watch`

**Comment by Shinigami92** [comment]
https://github.com/vitest-dev/vitest/pull/2612#issuecomment-1377797116

**Comment by matthewmayer** [comment]
could we make CI=true the default for the time being? to avoid scaring off new contributors who might not understand why the tests are not working.

**Comment by Shinigami92** [comment]
> could we make CI=true the default for the time being? to avoid scaring off new contributors who might not understand why the tests are not working.

I would prefer to put some pressure on https://github.com/vitest-dev/vitest/pull/2612 so the need is there

**Comment by matthewmayer** [comment]
hmm i was fairly convinced i had broken my local setup for a week or so as the tests kept failing because i didn't know about this issue, which discouraged me from working on faker for a while. 

Maybe you could add a message when `pnpm run test` is run without the flag, like "Due to a bug in vitest, tests may run very slowly. If your test run gets stuck, rerun with `CI=true pnpm run test`"

**Comment by Shinigami92** [comment]
It was more like "I do not want to put effort when the root cause will already get fixed in next few days"
That effort needs to be reverted when the new reporter has landed which results in double effort

**Comment by ST-DDT** [comment]
Fixed by #1762
